pkg/sloop/server/internal/config/config.go

```diff
@@ -75,77 +75,130 @@ type SloopConfig struct {
+	defaultConfig:=SloopConfig{
+		ConfigFile:               "",
+		DisableKubeWatcher:       false,
+		KubeWatchResyncInterval:  30*time.Minute,
+		WebFilesPath:             "./pkg/sloop/webserver/webfiles",
+		BindAddress:              "",
+		Port:                     8080,
+		StoreRoot:                "./data",
+		MaxLookback:              time.Duration(14*24)*time.Hour,
+		MaxDiskMb:                32*1024,
+		DebugPlaybackFile:        "",
+		DebugRecordFile:          "",
+		DeletionBatchSize:        1000,
+		UseMockBadger:            false,
+		DisableStoreManager:      false,
+		CleanupFrequency:         time.Minute*30,
+		KeepMinorNodeUpdates:     false,
+		DefaultNamespace:         "default",
+		DefaultKind:              "_all",
+		DefaultLookback:          "1h",
+		UseKubeContext:           "",
+		DisplayContext:           "",
+		ApiServerHost:            "",
+		WatchCrds:                true,
+		CrdRefreshInterval:       time.Duration(5*time.Minute),
+		ThresholdForGC:           0.8,
+		RestoreDatabaseFile:      "",
+		BadgerDiscardRatio:       0.99,
+		BadgerVLogGCFreq:         time.Minute*1,
+		BadgerMaxTableSize:       0,
+		BadgerLevelOneSize:       0,
+		BadgerLevSizeMultiplier:  0,
+		BadgerKeepL0InMemory:    true,
+		BadgerVLogFileSize:       0,
+		BadgerVLogMaxEntries:     200000,
+		BadgerUseLSMOnlyOptions:  true,
+		BadgerEnableEventLogging: false,
+		BadgerNumOfCompactors:    0,
+		BadgerNumL0Tables:        0,
+		BadgerNumL0TablesStall:   0,
+		BadgerSyncWrites:         true,
+		BadgerVLogFileIOMapping:  false,
+		BadgerVLogTruncate:       true,
+		EnableDeleteKeys:         false,
+	}
+	return &defaultConfig
+}
 // This will first check if a config file is specified on cmd line using a temporary flagSet
 // If not there, check the environment variable
 // If we have a config path, load initial values from it
 // Next parse flags again and override any fields from command line
 //
 // We do this to support settings that can come from either cmd line or config file
 func Init() *SloopConfig {
-	newConfig := &SloopConfig{}
+	finalConfig := registerDefaultSloopConfig()
+
+	configFileName:=""
+	configFileFlag := preParseConfigFlag()
+	configFileOS := os.Getenv(sloopConfigEnvVar)
 
-	configFilename := preParseConfigFlag()
-	glog.Infof("Config flag: %s", configFilename)
-	if configFilename == "" {
-		configFilename = os.Getenv(sloopConfigEnvVar)
-		glog.Infof("Config env: %s", configFilename)
+	if configFileFlag != "" {
+		configFileName = configFileFlag
+		glog.Infof("Config flag: %s", configFileFlag)
+	} else if configFileOS!=""{
+		configFileName=configFileOS
+		glog.Infof("Config env: %s", configFileOS)
 	}
-	if configFilename != "" {
-		newConfig = loadFromFile(configFilename)
-	} else {
-		registerFlags(flag.CommandLine, newConfig)
-		flag.Parse()
+	if configFileName != "" {
+		finalConfig = loadFromFile(configFileName,finalConfig)
 	}
+	finalConfig=registerFlags(flag.CommandLine,finalConfig)
```

Comment:
I don't think we need to return config from registerFlags.

Reply:
+1